base/compiler/ssair/ir.jl

```diff
@@ -147,7 +147,22 @@ function first_insert_for_bb(code, cfg, block)
 end
 
 
-const NewNode = Tuple{Int, #= reverse affinity =#Bool, Any, Any, #=LineNumber=#Int}
+struct NewNode
+    # Insertion position (interpretation depends on which array this is in)
+    pos::Int
+    # For BB purposes, attaches to the BB of the instruction before `pos`
+    reverse_affinity::Bool
+    # The type of the instruction to insert
+    typ::Any
+    # The node itself
+    node::Any
+    # The index into the line number table of this entry
+    line::Int
+end
+function NewNode(nnode::NewNode; pos=nnode.pos, reverse_affinity=nnode.reverse_affinity,
+                                 typ=nnode.typ, node=nnode.node, line=nnode.line)
```

Comment:
don't use named tuples for this – performance and compiler correctness will be really bad

Reply:
This is already fixed on master - I just haven't rebased this change since.